src/alto-ui/Pagination/Pagination.js

```diff
@@ -1,119 +1,96 @@
-        className={bemClass('Pagination__button', { arrow: true })}
-        onClick={() => onClick(currentValue + 1)}
-        disabled={currentValue === maxValue}
-      >
-        <ChevronRightIcon />
-      </button>
-    </div>
-  );
-};
-
-Pagination.displayName = 'Pagination';
-
-Pagination.defaultProps = {
-  className: '',
-};
-
-Pagination.propTypes = {
-  id: PropTypes.string,
-  /** additonal class to add to the wrapper component */
-  className: PropTypes.string,
-  /** current page number (between 1 -> max) */
-  current: PropTypes.number.isRequired,
-  /** max number of pages */
-  max: PropTypes.number.isRequired,
-  /** function that will be called each time the page change, first argument is the page number */
-  onClick: PropTypes.func.isRequired,
-  labels: PropTypes.shape({
-    pageLabel: PropTypes.string.isRequired,
-    currentPageLabel: PropTypes.string.isRequired,
-    goToPageLabel: PropTypes.string.isRequired,
-  }),
-};
-
-export default Pagination;
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { bemClass } from '../helpers/bem';
+import ChevronLeftIcon from '../Icons/ChevronLeft';
+import ChevronRightIcon from '../Icons/ChevronRight';
+import { format } from '../helpers/number';
+
+import './Pagination.scss';
+
+const DEFAULT_PAGE_SIZE = 10;
+
+const getRangeStart = (currentPage, pageSize, numberOfPages) => {
+  if (currentPage === 1) return 1;
+  if (currentPage >= numberOfPages) return (numberOfPages - 1) * pageSize + 1;
+  return (currentPage - 1) * pageSize + 1;
+};
+
+const getRangeEnd = (currentPage, numberOfPages, pageSize, total) => {
+  if (currentPage >= numberOfPages) return total;
+  return currentPage * pageSize;
+};
+
+const Pagination = ({ id, className, total, currentPage, onChange, pageSize, labels }) => {
+  const totalValue = Math.max(0, total);
+  if (totalValue <= 0) return null;
+  const currentPageValue = Math.min(total, Math.max(1, currentPage));
+  const numberOfPages = Math.ceil(totalValue / pageSize);
+  const rangeBeginIndex = getRangeStart(currentPageValue, pageSize, numberOfPages);
+  const rangeEndIndex = getRangeEnd(currentPageValue, numberOfPages, pageSize, totalValue);
+
+  const paginationLabels = {
+    pageLabel: 'page',
+    currentPageLabel: 'Current page',
+    goToPageLabel: 'Go to page',
+    ofTotal: 'of',
+    ...labels,
+  };
+
+  return (
+    <div id={id} className={bemClass('Pagination', {}, className)}>
+      <ChevronLeftIcon
+        id={id ? `${id}__button--prev` : undefined}
+        onClick={() => onChange(currentPage - 1)}
+        className={bemClass('Pagination__button', { arrow: true })}
+        disabled={currentPageValue === 1}
+        aria-label={`${paginationLabels.currentPageLabel}, ${
+          paginationLabels.pageLabel
+        } ${currentPageValue}`}
+      />
+      <span className="Pagination__range-items">{`${rangeBeginIndex}-${rangeEndIndex}`}</span>
+      <span className="Pagination__range-total">

```

Comment:
those `span` are not usefull

Reply:
Not exactly, in the design the range font is bolded, so I need some wrapper to bold font.
+ this is really helpful in tests.